apps/api/src/database/repositories/agent-repository.ts

```diff
@@ -98,6 +131,67 @@ export async function findByApiKey(
   }
 }
 
+/**
+ *
+ */
+export async function findByWallet({
+  walletAddress,
+  pagingParams,
+}: {
+  walletAddress: string;
+  pagingParams: PagingParams;
+}): Promise<SelectAgent[]> {
+  try {
+    let query = db
+      .select()
+      .from(agents)
+      .where(eq(agents.walletAddress, walletAddress))
+      .$dynamic();
+
+    if (pagingParams.sort) {
+      query = getSort(query, pagingParams.sort, agentOrderByFields);
+    }
+
+    query = query.limit(pagingParams.limit).offset(pagingParams.offset);
+
+    return await query;
+  } catch (error) {
+    console.error("[AgentRepository] Error in findByWallet:", error);
+    throw error;
+  }
+}
+
+/**
+ *
```

Comment:
(and might as well add the tsdoc here)

Reply:
Good call.  The tsdoc got lost in the rebase.